frontend/src/components/pages/your_clubs/CreateNewClub.tsx

```diff
@@ -0,0 +1,270 @@
+  const [missingField, setMissingField] = React.useState<boolean>(false);
+  const [submitSuccess, setSubmitSuccess] = React.useState<boolean>(false);
+  const [contentWarningsDisplay, setContentWarningsDisplay] =
+    React.useState<string|undefined>(undefined);
+
+  const contextServices = React.useContext(ServiceContext);
+  const yourClubsHandlerService = contextServices.yourClubsHandlerService;
+
+  const handleClubNameChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setClub({...club, name: event.target.value});
+  }
+
+  const handleClubDescriptionChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setClub({...club, description: event.target.value});
+  }
+
+  const handleClubContentWarningsChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    /*
+     * User should see newline characters in their repsonse,
+     * but backend must be stored as a list of strings.
+     */
+    setContentWarningsDisplay(event.target.value);
+    const contentWarningsArray = (event.target.value).split('\n');
+    setClub({...club, contentWarnings: contentWarningsArray});
+  }
+
+  const handleBookTitleChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setBook({...book, title: event.target.value});
+    setClub({...club, currentBook: book});
+  }
+
+  const handleBookAuthorChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setBook({...book, author: event.target.value});
+    setClub({...club, currentBook: book});
+  }
+
+  const handleBookIsbnChange = (event: React.ChangeEvent<HTMLInputElement>) => {
+    setBook({...book, isbn: event.target.value});
+    setClub({...club, currentBook: book});
+  }
+
+  const handleCloseCreate = (successfulCreation: boolean) => {
+    setClub(undefined);
+    setBook(undefined);
+    setContentWarningsDisplay(undefined);
+    setMissingField(false);
+    setSubmitSuccess(false);
+    props.closeWindow(successfulCreation);
+  }
+
+  const handleClubSubmission = async () => {
+    if (club && book
+          ? !(club.name && club.description && book.title && book.author)
+          : true) {
+      setMissingField(true);
+      setSubmitSuccess(false);
+    } else {
+      setMissingField(false);
+      const success = yourClubsHandlerService.createClub(club);
+      if (success) {
+        setSubmitSuccess(true);
+      }
+    }
+  }
+
+  return (
+    <div>
+      <Dialog open={props.isCreatingNewClub} maxWidth="lg">
+        <DialogTitle>Create New Club</DialogTitle>
+        <DialogContent>
+          <DialogContentText>
+            {submitSuccess
+              ? <b>Submission successful.</b>
+              : "By filling out the below fields and clicking 'Submit', a club \
+                 of which you have administrative rights over will be created." }
+          </DialogContentText>
+          <form>
+            <TextField
+              error = {
+                missingField
+                  ? (club ? (club.name ? false : true) : true)
```

Comment:
Multiple times in this code you first need to check if an object is defined, before checking if a member of that object is defined. You can compress many of this ternary operators using optional chaining (see https://www.typescriptlang.org/docs/handbook/release-notes/typescript-3-7.html#optional-chaining). We're working on TypeScript 3.9.5 so we should have access to this feature, but being honest I don't have much experience with it.

Reply:
It's my favorite recent TS feature, makes this kind of code so much cleaner :)

Reply:
Thank you thank you thank you for this pointer!

Reply:
```suggestion
                  missingField && !club?.name
```